go/vt/vtgate/engine/route.go

```diff
@@ -378,45 +381,99 @@ func (route *Route) paramsAllShards(vcursor VCursor, bindVars map[string]*queryp
 }
 
 func (route *Route) paramsSystemQuery(vcursor VCursor, bindVars map[string]*querypb.BindVariable) ([]*srvtopo.ResolvedShard, []map[string]*querypb.BindVariable, error) {
+	destinations, err := route.routeInfoSchemaQuery(vcursor, bindVars)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return destinations, []map[string]*querypb.BindVariable{bindVars}, nil
+}
+
+func (route *Route) routeInfoSchemaQuery(vcursor VCursor, bindVars map[string]*querypb.BindVariable) ([]*srvtopo.ResolvedShard, error) {
+	defaultRoute := func() ([]*srvtopo.ResolvedShard, error) {
+		ks := route.Keyspace.Name
+		destinations, _, err := vcursor.ResolveDestinations(ks, nil, []key.Destination{key.DestinationAnyShard{}})
+		return destinations, vterrors.Wrapf(err, "failed to find information about keyspace `%s`", ks)
```

Comment:
should we not check for error. returning destinations and also returning an error. It this intentional?

Reply:
yes, this is intentional. `vterrors.Wrap` will return `nil` if the error being wrapped is nil